modules/twinklespeedy.js

```diff
@@ -2027,9 +2027,18 @@ Twinkle.speedy.callback.evaluateUser = function twinklespeedyCallbackEvaluateUse
 	});
 
 	// analyse each criterion to determine whether to watch the page/notify the creator
-	var watchPage = normalizeds.some(function(norm) {
-		return Twinkle.getPref('watchSpeedyPages').indexOf(norm) !== -1 && Twinkle.getPref('watchSpeedyExpiry');
+	var watchPage = normalizeds.some(function(csdCriteria) {
+		var isWatchingThisCriteria = Twinkle.getPref('watchSpeedyPages').indexOf(csdCriteria) !== -1;
+		var isWatchingAllCSDs = Twinkle.getPref('watchSpeedyExpiry') !== 'no';
+		var shouldWatchThis = isWatchingThisCriteria && isWatchingAllCSDs;
+		return shouldWatchThis;
 	});
+
+	// if we should watch the page, replace "true" with the duration we should watch for
+	if (watchPage) {
+		watchPage = Twinkle.getPref('watchSpeedyExpiry');
+	}
+
```

Comment:
I think this is equivalent to just 
```js
	var watchPage = normalizeds.some(function(csdCriteria) {
		return Twinkle.getPref('watchSpeedyPages').indexOf(csdCriteria) !== -1;
	}) && Twinkle.getPref('watchSpeedyExpiry');
```
?
Twinkle.getPref('watchSpeedyExpiry') doesn't need to be checked every time in the loop, and if normalizeds.some returns true, watchSpeedyExpiry being `'no'` gives the desired effect